internal/helpers.go

```diff
@@ -61,3 +63,18 @@ func HostSystemManagementIPs(config []types.VirtualNicManagerNetConfig) []net.IP
 
 	return ips
 }
+
+// UsingVCEnvoySidecar determines if the given *vim25.Client is using vCenter's
+// local Envoy sidecar (as opposed to using the HTTPS port.)
+// Returns a boolean indicating whether to use the sidecar or not, and the sidecar port.
+func UsingVCEnvoySidecar(c *vim25.Client) (bool, string) {
```

Comment:
Does port need to be returned, since it's the same as caller's c.URL().Port() when true?
Also minor, but maybe just URL as the input param?
And super minor, but env vars don't have 'VC' in the name, reads better w/o it in the function name too:
```go
func UsingEnvoySidecar(u net.URL) bool
```

Reply:
Ah, good point. This also reminded me that it's preferable to use u.Host instead of Hostname():port, because the latter will break IPv6.